libs/wingc/src/type_check.rs

```diff
@@ -1245,11 +1245,15 @@ impl<'a> TypeChecker<'a> {
 
 				// Lookup the class's type in the env
 				let type_ = self.resolve_type_annotation(class, env, statement_idx);
-				let (class_env, class_symbol) = match &*type_ {
-					Type::Class(ref class) => (&class.env, &class.name),
+				let (class_env, class_symbol, has_fields) = match &*type_ {
+					Type::Class(ref class) => {
+						let fields = class.fields(true).into_iter().count();
+						(&class.env, &class.name, if fields > 0 { true } else { false })
+					}
 					Type::Resource(ref class) => {
 						if matches!(env.phase, Phase::Preflight) {
-							(&class.env, &class.name)
+							let fields = class.fields(true).into_iter().count();
+							(&class.env, &class.name, if fields > 0 { true } else { false })
```

Comment:
perhaps better
```suggestion
							let fields = class.fields(true).into_iter().count();
							(&class.env, &class.name, fields > 0)
```
or even
```suggestion
							(&class.env, &class.name, class.fields(true).peekable().peek().is_some())
```